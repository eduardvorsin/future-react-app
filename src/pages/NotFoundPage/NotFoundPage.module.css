.not-found{

}

.nout-found__title{
	
}
