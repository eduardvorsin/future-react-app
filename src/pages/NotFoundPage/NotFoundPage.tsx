import React from 'react';
import Title from '../../components/UI/Title/Title';
import classes from './NotFoundPage.module.css';

const NotFoundPage = () => (
  <div
    className={classes['not-found']}
  >
    <Title
      className={classes['nout-found__title']}
      variant='dark'
      level={4}
      component={'h1'}
    >
      404 Not Found
    </Title>
  </div>
);

export default NotFoundPage;
