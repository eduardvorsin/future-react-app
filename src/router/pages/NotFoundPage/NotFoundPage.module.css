.not-found {
	padding: 30px 0;
	background-image: url(../../../assets/images/not-found-bg.webp);
	background-color: var(--overlay-color-100-40);
	background-position: 0 0;
	background-size: cover;
	background-repeat: no-repeat;
	background-blend-mode: color;
	height: 100vh;
	min-height: 400px;
}

.not-found__container {
	display: flex;
	align-items: center;
	justify-content: center;
	flex-direction: column;
	text-align: center;
	height: 100%;
}

.not-found__title {
	color: var(--light-color-200);
	margin-bottom: 30px;
}

.not-found__title>span {
	display: block;
}

.not-found__link {
	font-size: 18px;
	color: var(--light-color-100);
	transition:
		color var(--transition-time) var(--transition-easing);
}

.not-found__link:hover {
	color: var(--brand-color-100);
	transition:
		color var(--transition-time) var(--transition-easing);
}