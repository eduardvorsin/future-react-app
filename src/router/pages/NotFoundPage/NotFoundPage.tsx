import React from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import Title from '../../../components/UI/Title/Title';
import classes from './NotFoundPage.module.css';

const NotFoundPage = () => {
  const { t } = useTranslation();
  return (
    <div
      className={classes['not-found']}
    >
      <div className={`${classes['not-found__container']} container`}>
        <Title
          className={classes['not-found__title']}
          level={1}
          component='h1'
        >
          <span>404</span>
          {t('mainTitle', { ns: 'notFoundPage' })}
        </Title>
        <Link
          className={classes['not-found__link']}
          to='/books'
        >
          {t('backToHomePage', { ns: 'notFoundPage' })}
        </Link>
      </div>
    </div>
  );
};

export default NotFoundPage;
