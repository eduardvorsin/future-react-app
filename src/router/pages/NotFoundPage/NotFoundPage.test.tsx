import React from 'react';
import { render, screen } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import NotFoundPage from './NotFoundPage';
import i18n from '../../../localization/i18next';

const router = createMemoryRouter([{
  path: '/',
  element: (
    <NotFoundPage
      testId='test-not-found-page'
    />
  ),
}]);

describe('NotFoundPage tests', () => {
  it('is rendered correctly', () => {
    render(<RouterProvider router={router} />);

    expect(screen.getByTestId<HTMLDivElement>('test-not-found-page')).toBeInTheDocument();
  });

  it('is a basic snapshot', () => {
    render(<RouterProvider router={router} />);

    expect(screen.getByTestId<HTMLDivElement>('test-not-found-page')).toMatchSnapshot();
  });
});

describe('BooksSearch integration tests', () => {
  it('the component with the text in English is displayed correctly', () => {
    const localizedTitle = i18n.t('mainTitle', { ns: 'notFoundPage' });

    render(<RouterProvider router={router} />);

    expect(screen.getByText<HTMLHeadingElement>(localizedTitle)).toBeInTheDocument();
  });

  it('the component with the text in Russian is displayed correctly', () => {
    i18n.changeLanguage('ru');
    const localizedTitle = i18n.t('mainTitle', { ns: 'notFoundPage' });

    render(<RouterProvider router={router} />);

    expect(screen.getByText<HTMLHeadingElement>(localizedTitle)).toBeInTheDocument();
  });
});
