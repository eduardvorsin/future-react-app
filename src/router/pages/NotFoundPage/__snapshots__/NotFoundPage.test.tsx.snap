// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage tests is a basic snapshot 1`] = `
<div
  class="not-found"
  data-testid="test-not-found-page"
>
  <div
    class="not-found__container container"
  >
    <h1
      class="title not-found__title title--dark title--level-1"
    >
      <span>
        404
      </span>
      Page Not Found
    </h1>
    <a
      class="not-found__link"
      href="/books"
    >
      Back to Main Page
    </a>
  </div>
</div>
`;
